Python/Python.sublime-syntax

```diff
@@ -1045,9 +1046,11 @@ contexts:
     - match: |- # simple form
         (?x)
         (\{)
-          (?: [\w.\[\]]+)?         # field_name
-          (   ! [ars])?            # conversion
-          (   : {{format_spec}} )? # format_spec
+          (?: [\w.\[\]]+)?             # field_name
+          (   ! [ars])?                # conversion
+          (   : (?:{{format_spec}}|    # format_spec OR
+                   (?:[^}\n]*?%.)+?)   # any format-like string
```

Comment:
The problem with this regex is that it will only match when the `%.` pattern matches at the end. I suggest adding another `[^}]*` after it. `\n` is redundant because then the terminating `\}` would cause the match to fail anyway as there can't be any more characters after a line break. The quantifier after the group can be dropped as well then as the intend is to match at least one `%.` pattern, not every. The new 'format-like string' pattern should thus be:

```regex
[^}%]*%.[^}]*
```

The following code/test case should be able to show the issue:
```py
'{0:%Y-%m-%dT}'
```

Reply:
Good points, certainly. It doesn't (and shouldn't?) highlight the technically-legal but rather boring `'{0:T}'`.